src/txHashBuilder.h

```diff
@@ -91,9 +115,27 @@ typedef struct {
 	};
 
 	tx_hash_builder_state_t state;
+	tx_hash_builder_output_substate_t subState;
```

Comment:
This smells of bad design --- a variable named "subState" should be generic, not related to just outputs. We can rename it (e.g. outputSubstate, and perhaps put it in an unnamed union as the one above, for now the union will have just a single member), but we need to think about how it is related to the union above --- maybe some of its items actually belong to the substate? I guess we don't want to overcomplicate this, but also don't want to have confusing names/structures.

Reply:
In sign tx  `sign_tx_output_state_t` is used, maybe I can get rid of "sub" and make it as :
tx_hash_builder_output_state_t outputState?

Reply:
I guess yes, I'm not sure what you refer to.